docs/testing-guide.md

```
@@ -0,0 +1,472 @@
+
+test('ComponentName - should handle edge cases', t => {
+  // Test implementation
+  t.true(condition);
+});
+```
+
+### 2. Mocking Strategy
+
+#### Common Patterns
+
+**Sinon Stubs**
```

markdownlint-cli2 (0.17.2)

Emphasis used instead of a heading

(MD036, no-emphasis-as-heading)